### PR TITLE
Extend `IrGenerators` with function operators

### LINKED ISSUE #1355: [FEATURE] Add function operators to IrGenerators
Extend `IrGenerators` to also apply function operators. This would be useful in property-based tests such as the ones added in #1344. Original request: https://github.com/informalsystems/apalache/pull/1344#discussion_r807066075

- [x] Extend `IrGenerators` to also apply function operators
- [x] Use them for PBT in `TestKeramelizer`
- [x] Identify other places these can be used

### PR DESCRIPTION
**Pull request by thpani**
Closes #1355

* Extends `IrGenerators` with function operators.
* Adds them to PBT specs where applicable.

### CODE REVIEW

File: tlair/src/test/scala/at/forsyte/apalache/tla/lir/IrGenerators.scala [hunk 1]
```diff
@@ -80,6 +80,12 @@ trait IrGenerators extends TlaType1Gen {
       TlaSetOper.in, TlaSetOper.notin, TlaSetOper.map, TlaSetOper.powerset, TlaSetOper.recSet, TlaSetOper.seqSet,
       TlaSetOper.setminus, TlaSetOper.subseteq, TlaSetOper.times, TlaSetOper.union)
 
+  /**
+   * Function operators (`TlaFunOper._`)
+   */
+  val functionOperators = List(TlaFunOper.enum, TlaFunOper.tuple, TlaFunOper.app, TlaFunOper.domain, TlaFunOper.funDef,
+      TlaFunOper.recFunDef, TlaFunOper.recFunRef, TlaFunOper.except)
```
Comment (Kukovec) [inline tlair/src/test/scala/at/forsyte/apalache/tla/lir/IrGenerators.scala:87]: I think there's very limited value in supporting `recFunDef` and `recFunRef`. Not only are they the next to go, they will also generate countless malformed expressions, due to the restrictions on their proper use.

**Review by Kukovec — APPROVED** [review]
LGTM

File: tlair/src/test/scala/at/forsyte/apalache/tla/lir/IrGenerators.scala (hunk 1 above)
Comment (konnov) [reply tlair/src/test/scala/at/forsyte/apalache/tla/lir/IrGenerators.scala:87]: I am not certain yet about letting the recursive functions to go (in contrast to recursive operators). I have seen some value in them recently. Though they can always be replaced with `CHOOSE` over a function set.

**Review by konnov — APPROVED** [review]
Looks good!

File: tlair/src/test/scala/at/forsyte/apalache/tla/lir/IrGenerators.scala (hunk 1 above)
Comment (thpani) [reply tlair/src/test/scala/at/forsyte/apalache/tla/lir/IrGenerators.scala]: Tested this on the Keramelizer PBT with `minSuccessful = 10,000` and it didn't produce a malformed expression, so I think it's fine in this regard. If `reckFun{Def,Ref}` become obsolete, we can clean them out then.

**Pull request merged**